pkgs/development/tools/build-managers/meson/default.nix

```diff
@@ -70,12 +75,19 @@ python3.pkgs.buildPythonApplication rec {
       url = "https://github.com/mesonbuild/meson/commit/a38ad3039d0680f3ac34a6dc487776c79c48acf3.patch";
       hash = "sha256-9YaXwc+F3Pw4BjuOXqva4MD6DAxX1k5WLbn0xzwuEmw=";
     })
-  ];
+  ]
+    # Nixpkgs cctools does not have bitcode support.
+    ++ lib.optional stdenv.isDarwin ./disable-bitcode.patch;
+
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    rm -r 'test cases/osx/7 bitcode'
+  '';
```

Comment:
optionalString creates an empty string, which is different from null and causes rebuilds (my memory is vague on this, correct me if I'm wrong)

Reply:
Or prepend it to checkPhase? Anyway we have to fix it before the next cycle.